src/App/src/Middleware/ContentNegotiationMiddleware.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Api\App\Middleware;
+
+use Dot\AnnotatedServices\Annotation\Inject;
+use Laminas\Diactoros\Response\JsonResponse;
+use Laminas\Http\Response;
+use Mezzio\Router\RouteResult;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+use function array_filter;
+use function array_intersect;
+use function array_map;
+use function explode;
+use function in_array;
+use function is_array;
+use function str_contains;
+use function strtok;
+use function trim;
+
+class ContentNegotiationMiddleware implements MiddlewareInterface
+{
+    /**
+     * @Inject({"config.content-negotiation"})
+     */
+    public function __construct(private array $config)
+    {
+    }
+
+    public function process(
+        ServerRequestInterface $request,
+        RequestHandlerInterface $handler
+    ): ResponseInterface {
+        /** @var RouteResult $routeResult */
```

Comment:
```suggestion
```
Because this line:
```php
if (! $routeResult instanceof RouteResult || $routeResult->isFailure()) {
```
both makes sure that `$routeResult` is an instance of `RouteResult` and helps the IDE with autocomplete.